### PR TITLE
Keep puppetclasses selected when user changes puppet environment

### LINKED ISSUE #23: puppetclass_ids are removed when selecting a different puppet-environment
### Expected behaviour:
When default_values.puppetclass_ids is defined and the user selects a different puppet environment, the defined Puppet classes keeps automatically assigned to the new host.

### Actual behaviour:
The classes are removed and the user needs to select them manually.

### PR DESCRIPTION
**Pull request by kamils-iRonin**
Fixes #23

### CODE REVIEW

File: src/components/HostsForm/PuppetConfig/Custom/PuppetEnvSelectInput.js [hunk 1]
```diff
@@ -1,26 +1,35 @@
...
+    .map(({
+      node: {
+        id,
+        name,
+        puppetclasses: { edges: puppetclasses }
+      }
+    }) => ({ id, name, puppetclasses: puppetclasses.map(({ node: { id }}) => id) }))
+
+  const currentPuppetEnv = find(puppetEnvs, ['id', puppetEnvId]) || {}
+  const newPuppetclassIds = puppetclassIds.filter((id) => get(currentPuppetEnv, 'puppetclasses', []).includes(id))
```
Comment (kamils-iRonin) [inline src/components/HostsForm/PuppetConfig/Custom/PuppetEnvSelectInput.js:29]: we need to filter out puppet classes that do not belong to the selected puppet environment

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -9,7 +9,7 @@ jobs:
       fail-fast: true
       max-parallel: 2
       matrix:
-        node-version: [8.x, 12.x, 14.x]
+        node-version: [12.x, 14.x, 16.x]
```
Comment (kamils-iRonin) [inline .github/workflows/ci.yml:12]: do we need to support nodejs 8?

Comment (laugmanuel) [reply .github/workflows/ci.yml:12]: I don't think so as its EOL for over two years now.

Comment (kamils-iRonin) [reply .github/workflows/ci.yml:12]: 👍  just merged https://github.com/dm-drogeriemarkt/lisa/pull/52

**Comment by laugmanuel** [comment]
I had a look at this PR.
The feature generally works but the initial request (collecting all Puppetservers and envs) takes ages (32sec in my case). Before this MR, the call took "only" 3sec.

Can we call the API asynchronously?

**Comment by kamils-iRonin** [comment]
@laugmanuel can you please check the performance after the last changes?

**Comment by laugmanuel** [comment]
I'm using the following commit:
```sh
❯ git rev-parse HEAD
991e842fdf96dfe5066a253c13cb9a0a6fe15d59
```

But it's still very slow on the initial request:
image
Subsequent requests are fast again and keep the class selected.

**Comment by kamils-iRonin** [comment]
@laugmanuel the performance should now be unaffected

**Comment by laugmanuel** [comment]
> @laugmanuel the performance should now be unaffected

Can confirm that performance is great now and the feature works as expected!

**Review by laugmanuel — APPROVED** [review]

**Pull request merged**